File: app/src/main/java/com/hlibrary/utils/ActivityManager.java
package com.hlibrary.utils;


import com.hlibrary.base.activity.BaseActivity;

import java.util.TreeMap;


/**
 * @author linwh
 * @date 2015-09-06
 */
public class ActivityManager {

    private TreeMap<String, BaseActivity> activities;


    public ActivityManager() {
        activities = new TreeMap();
    }

    public void pushActivity(BaseActivity baseActivity) {
        activities.put(baseActivity.getClass().getSimpleName(), baseActivity);
    }

    public void pullActivity(BaseActivity baseActivity) {
        activities.remove(baseActivity.getClass().getSimpleName());
    }

    public BaseActivity getCurActivity() {
        return activities.lastEntry().getValue();
    }

    public String getCurActivityName() {
        if (activities.isEmpty())
            return "";
        return activities.lastKey();
    }

    public void exit() {
        for (String key : activities.keySet()) {
            BaseActivity baseActivity = activities.remove(key);
            baseActivity.finish();
        }
    }

}

File: app/src/main/java/com/hlibrary/utils/ActivityManager.kt
package com.hlibrary.utils

import com.hlibrary.base.activity.BaseActivity
import java.util.*

/**
 * @author linwh
 * @date 2015-09-06
 */
class ActivityManager private constructor() {
    private val activities: TreeMap<String?, BaseActivity<*>?> = TreeMap()

    fun pushActivity(baseActivity: BaseActivity<*>) {
        activities[baseActivity.javaClass.simpleName] = baseActivity
    }

    fun popActivity(baseActivity: BaseActivity<*>) {
        activities.remove(baseActivity.javaClass.simpleName)
    }

    val curActivity: BaseActivity<*>?
        get() = activities.lastEntry().value

    val curActivityName: String?
        get() = if (activities.isEmpty()) {
            ""
        } else activities.lastKey()

    fun exit() {
        for (key in activities.keys) {
            val baseActivity = activities.remove(key)
            baseActivity!!.finish()
        }
    }

    companion object {
        val instance by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED){
            ActivityManager()
        }
    }


}